remotes/mount.go

```diff
@@ -150,17 +126,74 @@ func (r *remoteReaderAt) ReadAt(p []byte, off int64) (int, error) {
 	return n, err
 }
 
-type descriptorAccumulator struct {
-	descriptors []ocischemav1.Descriptor
+type manifestWalker struct {
+	getChildren       images.HandlerFunc
+	resolver          remotes.Resolver
+	targetRepo        string
+	eventNotifier     eventNotifier
+	descriptorHandler imageHandler
+	scheduler         scheduler
+	progress          *progress
+}
+
+func newManifestWalker(targetRepo string, resolver remotes.Resolver, getChildren images.HandlerFunc,
+	descriptorHandler imageHandler, eventNotifier eventNotifier, scheduler scheduler, progress *progress) *manifestWalker {
```

Comment:
That's a beast of an argument list. Not sure much can be done other than just having callers create the struct themselves and doing away with the wrapper though.

Reply:
yead, I am not very happy with this. I'll try to split it a little